modules/e2ee/OlmAdapter.js

```diff
@@ -470,6 +572,199 @@ export class OlmAdapter extends Listenable {
             }
             break;
         }
+        case OLM_MESSAGE_TYPES.SAS_START: {
+            if (olmData.session) {
+                if (olmData.sasVerification?.sas) {
+                    logger.warn(`SAS already created for participant ${pId}`);
+
+                    return;
+                }
+
+                const { uuid } = msg.data;
+
+                olmData.sasVerification.sas = new Olm.SAS();
+                olmData.sasVerification.uuid = uuid;
+
+                const pubKey = olmData.sasVerification.sas.get_pubkey();
+                const olmUtil = new Olm.Utility();
+                const commitment = olmUtil.sha256(pubKey + msg.data);
```

Comment:
When computing the commitment we need to JSON encode the data so it's a string. I suggest currently the commitment looks like `'lalala[object Object]'`

Reply:
Since we need to do it twice, maybe it would be a good idea to create a `_computeCommitment(pubKey: string, data: object) ` helper that takes care of this.